### DISCUSSION TITLE
Which events have what information

### DISCUSSION THREAD

**Discussion by HritwikSinghal** [microsoft/gctoolkit#218] [Q&A] [answered]
So while Reading the Developers guide, it said that the information an event contains is in its javadoc but many events does not have getDuration() method. Can anyone please help me where can i find what data an event stores since by inheritance, every Generational event should support getDuration() but many (like PSYoungGen) does not have that data and i also cant find this info in javadoc

**Comment by dsgrieve** [comment, marked as answer]
Every event inherits from com.microsoft.gctoolkit.event.jvm.JVMEvent which is where getDuration() is defined. Therefore, every event has a getDuration() method. 

I checked the generated javadoc for PSYoungGen and the methods inherited from JVMEvent are given. You may need to run `./mvnw javadoc:javadoc` to generate the javadoc. PSYoungGen is in the javadoc of the api module.

Of course, there is much that can be done to improve the javadoc.

**Reply by HritwikSinghal** [reply]
> Every event inherits from com.microsoft.gctoolkit.event.jvm.JVMEvent which is where getDuration() is defined. Therefore, every event has a getDuration() method.
> 
> I checked the generated javadoc for PSYoungGen and the methods inherited from JVMEvent are given. 

Yes, the method is inherited from JVMEvent. However that method does not return anything for PSYoungGen event. Similar is the case for many other events. By inheritence, it should have that data. But either the parsers are not able to capture those events, or there is something else wrong. 

For reference, i ran the same GC log file with GCToolkit and GCeasy. GCEasy was able to get PSYoungGen event duration, GCToolkit was not.

Let me know if you need the GC Log file.

**Reply by karianna** [reply]
@HritwikSinghal - I opened #224 - can you post the log in that issue?

**Comment by kcpeppe** [comment]
Hi,

I believe the parser are capturing this information as they have in the
past. I’m not sure of what has gone off to create this particular issue but
will dig into it as soon as I can.

—- Kirk

**Reply by HritwikSinghal** [reply]
hey, here is more info for debugging. 

https://github.com/microsoft/gctoolkit/issues/224#issuecomment-1178417559
